packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -199,6 +206,14 @@ class ConfigChainBuilder {
 
     const envKey = babel.getEnv();
     if (options.env) {
+      // TODO: Remove in Babel v8
+      if (!warnedDeprecatedEnv) {
+        console.log(`The "env" option has been deprecated and will be removed in a future version of Babel.
```

Comment:
Ah, it loos like that linked PR also touches the `env` logic, so this PR might not be what we want anyway.